consensus/src/round_manager.rs

```diff
@@ -1335,49 +1340,58 @@ impl RoundManager {
 
     async fn new_qc_from_order_vote_msg(
         &mut self,
-        order_vote_msg: &OrderVoteMsg,
+        qc: &QuorumCert,
+        order_vote_author: Author,
     ) -> anyhow::Result<()> {
-        if let NeedFetchResult::QCAlreadyExist = self
-            .block_store
-            .need_fetch_for_quorum_cert(order_vote_msg.quorum_cert())
-        {
+        if let NeedFetchResult::QCAlreadyExist = self.block_store.need_fetch_for_quorum_cert(qc) {
             return Ok(());
         }
 
         let start = Instant::now();
-        order_vote_msg
-            .quorum_cert()
-            .verify(&self.epoch_state().verifier)
+        qc.verify(&self.epoch_state().verifier)
             .context("[OrderVoteMsg QuorumCert verification failed")?;
         counters::VERIFY_MSG
             .with_label_values(&["order_vote_qc"])
             .observe(start.elapsed().as_secs_f64());
 
         let result = self
             .block_store
-            .insert_quorum_cert(
-                order_vote_msg.quorum_cert(),
-                &mut self.create_block_retriever(order_vote_msg.order_vote().author()),
-            )
+            .insert_quorum_cert(qc, &mut self.create_block_retriever(order_vote_author))
             .await
             .context("[RoundManager] Failed to process the QC from order vote msg");
         self.process_certificates().await?;
         result
     }
 
     // Insert ordered certificate formed by aggregating order votes
-    async fn new_ordered_cert(&mut self, ordered_cert: WrappedLedgerInfo) -> anyhow::Result<()> {
+    async fn new_ordered_cert(
+        &mut self,
+        ordered_cert: WrappedLedgerInfo,
+        qc: &QuorumCert,
+        order_vote_author: Author,
+    ) -> anyhow::Result<()> {
+        // If the block doesn't exist, we could ideally do sync up based on the qc.
+        // But this could trigger fetching a lot of past blocks in case the node is lagging behind.
+        // So, we just log a warning here to avoid a long sequence of block fetchs.
+        // One of the subsequence syncinfo messages will trigger the block fetch or state sync if required.
         if self
             .block_store
             .get_block(ordered_cert.commit_info().id())
             .is_none()
         {
             ORDER_CERT_CREATED_WITHOUT_BLOCK_IN_BLOCK_STORE.inc();
-            error!(
+            warn!(
                 "Ordered certificate created without block in block store: {:?}",
                 ordered_cert
             );
```

Comment:
this seems like normal operation - so should be a an info or debug. Also this can happen many times per round, so should be sampled - i.e. once every 30s. we have the counter to track the actual number of occurrences.